theano/sandbox/cuda/cuda_ndarray.cu

```diff
@@ -1026,13 +1026,21 @@ CudaNdarray_TakeFrom(CudaNdarray * self, PyObject *args){
                             " indices with only 1 dimensions");
             return NULL;
         }
+        // We need indices_obj to be contiguous, in order to take a view
+        // with a different dtype.
+        if (!PyArray_IS_C_CONTIGUOUS((PyArrayObject*) indices_obj)) {
```

Comment:
I would suggest that you use: tmp = PyArray_GETCONTIGUOUS(indices_obj)

Then always decref tmp when you don't need it anymore.